doc/release-notes/10.4.md

```diff
@@ -0,0 +1,36 @@
+# Release notes for Gluster 10.4
+
+This is a bugfix release. The release notes for [10.0](10.0.md), [10.1](10.1.md), [10.2](10.2.md) and [10.3](10.3.md) contain a listing of all the new improvements and bugs fixed in the GlusterFS 10 stable release.
+
+**NOTE:**
+- Next minor release tentative date: Week of 25th Sep, 2023
+- Users are highly encouraged to upgrade to newer releases of GlusterFS.
+
+## Important fixes in this release
+- Fix fuse concurrency problems
+- Fix memory corruption in debug builds
+- Fix recovery issue with posix locks upon reconnection of a disconnected brick
+
+## Builds are available at
+
+[https://download.gluster.org/pub/gluster/glusterfs/10/10.4/](https://download.gluster.org/pub/gluster/glusterfs/10/10.4/)
+
+## Issues addressed in this release
+
+- [#1000](https://github.com/gluster/glusterfs/issues/1000) configure: Force 'char' type to be signed inorder to eliminate anomalies
+- [#2752](https://github.com/gluster/glusterfs/issues/2752) posix: Fix directory gfid handle if a rename fails
+- [#3345](https://github.com/gluster/glusterfs/issues/3345) Fix inconsistencies in big-endian architectures of hashfn
+- [#3346](https://github.com/gluster/glusterfs/issues/3346) Fix stack overflow when processing glx_dir\(p\) list structures in xdr
+- [#3659](https://github.com/gluster/glusterfs/issues/3659) Fix fuse concurrency problems
+- [#3855](https://github.com/gluster/glusterfs/issues/3855) Fix memory corruption in debug builds
```

Comment:
The patch for this issue included several changes from devel to be able to easily backport the principal patch. However the other patches are mostly cleanups, so probably this entry is not required.